crates/storage/src/types.rs

```diff
@@ -62,9 +63,15 @@ pub mod state_update {
     #[serde(deny_unknown_fields)]
     pub struct StateDiff {
         pub storage_diffs: Vec<StorageDiff>,
-        pub declared_contracts: Vec<DeclaredContract>,
+        /// Refers to Declare V0 & V1 txns, these contain Cairo classes
+        pub declared_contracts: Vec<DeclaredCairoClass>,
+        /// Refers to pre-Starknet-0.11.0 Deploy txns
         pub deployed_contracts: Vec<DeployedContract>,
         pub nonces: Vec<Nonce>,
+        /// Refers to Declare V2 txns, these contain Sierra classes
+        pub declared_sierra_classes: Vec<DeclaredSierraClass>,
```

Comment:
I think we need `#[serde(default)]` here otherwise loading old state updates from storage would fail?